packages/lodestar/src/chain/chain.ts

```diff
@@ -159,20 +154,44 @@ export class BeaconChain extends (EventEmitter as { new(): ChainEventEmitter })
     });
     this.logger.info("Beacon chain initialized");
   }
-  
 
-  private checkGenesis = async (eth1Block: Block): Promise<void> => {
+  // If we don't have a state yet, we have to wait for genesis state
+  private async waitForState(): Promise<BeaconState> {
+    let state: BeaconState;
+    try {
+      state = await this.db.state.getLatest();
+    } catch (err) {
+      this.logger.info("Chain not started, listening for genesis block");
+      state = await new Promise((resolve) => {
+        this.eth1Listener = (eth1Block: Block): void => {
+          this.checkGenesis(eth1Block).then((state) => {
+            if (state) {
+              resolve(state);
+            }
+          });
+        };
+        this.eth1.on("block", this.eth1Listener);
+      });
+    }
+    if (this.eth1Listener) {
+      this.eth1.removeListener("block", this.eth1Listener);
+    }
+    return state;
+  }
+
+  private checkGenesis = async (eth1Block: Block): Promise<BeaconState> => {
     this.logger.info(`Checking if block ${eth1Block.hash} will form valid genesis state`);
-    const depositDatas = await this.opPool.depositData.getAll();
+    const depositDatas = await this.eth1.processPastDeposits(undefined, eth1Block.number);
+    this.logger.info(`Found ${depositDatas.length} deposits`);
     const depositDataRootList = this.config.types.DepositDataRootList.tree.defaultValue();
-    depositDataRootList.push(...depositDatas.map(this.config.types.DepositData.hashTreeRoot));
     const tree = depositDataRootList.tree();
 
     const genesisState = initializeBeaconStateFromEth1(
       this.config,
       fromHexString(eth1Block.hash),
       eth1Block.timestamp,
       depositDatas.map((data, index) => {
+        depositDataRootList.push(this.config.types.DepositData.hashTreeRoot(data));
```

Comment:
I can't verify deposit without this change (although the old/current looks more reasonable to me). I copied the way to get proof from https://github.com/ChainSafe/lodestar/blob/0.11.x/packages/lodestar-cli/src/lodestar/interop/deposits.ts#L33

any ideas?

Reply:
found it. That's because of the way we process deposits in `initializeBeaconStateFromEth1` so the proof should be calculated accordingly.